packages/kuma-gui/src/app/kuma/components/ApplicationShell.vue

```diff
@@ -117,33 +117,38 @@
               Onboarding
             </XAction>
           </XActionGroup>
-          <XAction
-            :to="{ name: 'diagnostics' }"
-            appearance="tertiary"
-            icon
-            data-testid="nav-item-diagnostics"
-          >
-            <XIcon
-              name="settings"
-            >
-              Diagnostics
-            </XIcon>
-          </XAction>
         </slot>
       </div>
     </header>
     <div
       class="app-content-container"
     >
-      <nav
-        v-if="$slots.navigation"
-        aria-label="Main"
-        class="app-sidebar"
-      >
-        <ul>
-          <slot name="navigation" />
-        </ul>
-      </nav>
+      <aside class="app-sidebar">
+        <div class="navigation-container">
+          <nav
+            v-if="$slots.navigation"
+            aria-label="Main"
+          >
+            <ul>
+              <slot name="navigation" />
+            </ul>
+          </nav>
+
+          <div
+            role="separator"
+            class="navigation-separator"
+          />
+
+          <nav
+            v-if="$slots.bottomNavigation"
+            aria-label="Secondary"
+          >
+            <ul>
+              <slot name="bottomNavigation" />
+            </ul>
+          </nav>
+        </div>
+      </aside>
```

Comment:
I have a few comments on this, most of them are "ideally":

- I'd say we shouldn't use an `aside` here. `aside` is generally used for content "indirectly related to the documents content". The important semantic tag here is the `nav` i.e. navigational links, I don't think we should re-wrap that again in something that says "this is content indirectly related to the documents content". Whilst its tempting to see a "sidebar" and think "aside", that's kinda mixing visuals/decorative concerns with meaning/semantic concerns.
- The second nav shouldn't be the "Secondary Navigation", "ideally" it would be part of the "Main Navigation", its just one more item in the "Main Navigation" the fact that visually its in a different place is kinda mixing the decorative/semantic concerns similar to point 1.

I think for the above two, sometimes its helpful to try to ignore the designs completely, just look at the HTML document and attempt to describe what the content is (not what it looks like) using HTML tags.

To try and give some alternative options:

1. "ideally" there is no difference between Home, Zones, Meshes and Configuration semantically. They are all part of the main navigation, the only reason you've split them here is because visually they are in a different place. Therefore "ideally" all of these items would be in the same `<nav aria-label="Main"><ul>` as separate `li`s. Then I would guess you "might" be able to apply similar CSS to the `ul` and the Configuration `li` to move that to the bottom (this might be overly difficult, if so see next idea)
2. We could also try keeping all the items in the `<nav aria-label="Main">` but putting Configuration in a separate `ul` to the other items, and again trying to apply similar CSS. This is cheating a little, but not as much as having 2 nav's.

Hopefully one of the above works, if not we will have to think of a different option that doesn't have a "Secondary Navigation" that isn't has links that aren't children of the "Main Navigation". Maybe we just name it something else.

Anyway let me know what you think, the entire thing is a bit of a nit really, but if we are aiming to do semantics we should try and do the best we can. If it proves problematic, we can just say "ok we tried to be super semantic here, but realistically it was overly difficult so we cheated"

Gonna submit this comment while I keep reviewing.

Reply:
> I have a few comments on this, most of them are "ideally":
> 
> * I'd say we shouldn't use an `aside` here. `aside` is generally used for content "indirectly related to the documents content". The important semantic tag here is the `nav` i.e. navigational links, I don't think we should re-wrap that again in something that says "this is content indirectly related to the documents content". Whilst its tempting to see a "sidebar" and think "aside", that's kinda mixing visuals/decorative concerns with meaning/semantic concerns.

hmm 🤔 I think that `aside` is also frequently used for sidebars. If we keep the semantics wherever possible I'd say they are more stable when it comes to refactorings. The `aside` is a container for anything that is, as you said, indirectly related to the documents context. Which is the case for nav links, but could also be other things. So if we decide to move the nav links somewhere else, the sidebar may stay there.

> * The second nav shouldn't be the "Secondary Navigation", "ideally" it would be part of the "Main Navigation", its just one more item in the "Main Navigation" the fact that visually its in a different place is kinda mixing the decorative/semantic concerns similar to point 1.

Yes, I fully agree. This makes more sense 👍

Reply:
> To try and give some alternative options:
> 
> 1. "ideally" there is no difference between Home, Zones, Meshes and Configuration semantically. They are all part of the main navigation, the only reason you've split them here is because visually they are in a different place. Therefore "ideally" all of these items would be in the same `<nav aria-label="Main"><ul>` as separate `li`s. Then I would guess you "might" be able to apply similar CSS to the `ul` and the Configuration `li` to move that to the bottom (this might be overly difficult, if so see next idea)
> 2. We could also try keeping all the items in the `<nav aria-label="Main">` but putting Configuration in a separate `ul` to the other items, and again trying to apply similar CSS. This is cheating a little, but not as much as having 2 nav's.

I went for option 2 🙂 👍

Reply:
> hmm 🤔 I think that aside is also frequently used for sidebars.

Would you say that if we have a different style of navigation, say a top horizontal nav like this:

![Screenshot 2024-11-22 at 10 24 18](https://github.com/user-attachments/assets/cc7c69a3-9531-4495-ab6d-a95d2b450642)

With `[Home] [Meshes] [Zones] [Configuration]` along the top horizontally,  then you wouldn't use an `aside` as its not at the side of the page?